model/task_lifecycle.go

```diff
@@ -1127,6 +1127,36 @@ func updateBuildGithubStatus(b *build.Build, buildTasks []task.Task) error {
 	return b.UpdateGithubCheckStatus(buildStatus.status)
 }
 
+// checkUpdateBuildPRStatus checks if the build is coming from a PR, and if so
+// sends its updated status to GitHub to reflect the status of the build.
```

Comment:
It seems like this function is specifically restricted to resetting the state back to pending, rather than updating the state arbitrarily. Can we update this doc comment to clarify that?